### PR TITLE
[3] don't remove emailcloack from module

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #13522.

### Summary of Changes
don't remove `<joomla-hidden-mail>`

### Testing Instructions

-   Add an "Articles Category" module on the sample site
-   Under "Display Options" tab: everything on Hide, except "Introtext= show" and "Show Read More=show"
-   Create an article with the below html:

```
<p>Intro text <a href="mailto:sample@example.com">sample@example.com</a> - more more more</p>
<hr id="system-readmore" />
<p><br />Paragraph 1<br />1<br />2<br />3<br />4<br />5</p>
<p>Paragraph 2<br />1<br />2<br />3<br />4<br />5</p>
<p>Paragraph 3<br />1<br />2<br />3<br />4<br />5</p>
<p>End</p>
```

### Actual result BEFORE applying this Pull Request
This email address is being protected from spambots. You need JavaScript enabled to... 

### Expected result AFTER applying this Pull Request

Introtext shows the email address

### CODE REVIEW

**Comment by jsubri** [comment]
I have tested this item :white_check_mark: successfully on ec6ea546c5a4fdad13b218b12a5d0b64d3e3af74<br /><br />Clever fix

Before the patch the javascript is render "document.getElementById('cloak1bb8d18fc01bde967325b35c999b2ffa').inner..."

After, works as expected

tested on 3.9.x and 3.10.4, not test tested on J4

Thank you
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36603">issues.joomla.org/tracker/joomla-cms/36603</a>.</sub>

File: modules/mod_articles_category/helper.php [hunk 1]
```diff
@@ -301,11 +301,13 @@ public static function getList(&$params)
...
 			{
-				$item->introtext = JHtml::_('content.prepare', $item->introtext, '', 'mod_articles_category.content');
+				
 				$item->introtext = self::_cleanIntrotext($item->introtext);
+				$item->introtext=  self::truncate($item->introtext, $introtext_limit);
+				$item->introtext = JHtml::_('content.prepare', $item->introtext, '', 'mod_articles_category.content');
 			}
 
-			$item->displayIntrotext = $show_introtext ? self::truncate($item->introtext, $introtext_limit) : '';
+			$item->displayIntrotext = $item->introtext;
```
Comment (Fedik) [inline modules/mod_articles_category/helper.php]: This always will be shown, even if "show_introtext" is Off

Comment (Fedik) [reply modules/mod_articles_category/helper.php]: change to:

```php
$item->displayIntrotext = $show_introtext ? $item->introtext : '';
```

Comment (jsubri) [reply modules/mod_articles_category/helper.php]: @Fedik well spoted

File: modules/mod_articles_category/helper.php [hunk 2]
```diff
@@ -301,11 +301,13 @@ public static function getList(&$params)
 
 			if ($show_introtext)
 			{
-				$item->introtext = JHtml::_('content.prepare', $item->introtext, '', 'mod_articles_category.content');
+
```
Comment (PhilETaylor) [inline modules/mod_articles_category/helper.php]:
```suggestion
```

File: modules/mod_articles_category/helper.php [hunk 3]
```diff
@@ -301,11 +301,12 @@ public static function getList(&$params)
 
 			if ($show_introtext)
 			{
-				$item->introtext = JHtml::_('content.prepare', $item->introtext, '', 'mod_articles_category.content');
 				$item->introtext = self::_cleanIntrotext($item->introtext);
+				$item->introtext=  self::truncate($item->introtext, $introtext_limit);
```
Comment (Quy) [inline modules/mod_articles_category/helper.php]:
```suggestion
				$item->introtext = self::truncate($item->introtext, $introtext_limit);
```

**Comment by jsubri** [comment]
I have tested this item :white_check_mark: successfully on 55e510328f9b53583ed2052d732c985f26b5c017<br /><br />Now respecting the show/hide Introtext pointed out by @Fedik 
Thank you<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36603">issues.joomla.org/tracker/joomla-cms/36603</a>.</sub>

**Comment by Fedik** [comment]
I just found, changing the code ordering may break more than fix.
Example if Article contain "loadmodule/loadposition" tag:
```
<p>{loadposition navigation}</p>
<p>Intro text <a href="mailto:sample@example.com">sample@example.com</a> - more more more</p>
<hr id="system-readmore" />
<p><br />Paragraph 1<br />1<br />2<br />3<br />4<br />5</p>
<p>Paragraph 2<br />1<br />2<br />3<br />4<br />5</p>
<p>Paragraph 3<br />1<br />2<br />3<br />4<br />5</p>
<p>End</p>
```
Then whole content will be rendered.

Need to find another way.

**Pull request closed without merging**